lib/change_stream.js

```diff
@@ -127,7 +127,31 @@ class ChangeStream extends EventEmitter {
    * @returns {Promise} returns Promise if no callback passed
    */
   hasNext(callback) {
-    return this.cursor.hasNext(callback);
+    // intercept and ignore errors during resume attempts
+    if (this.attemptingResume) {
+      return;
+    }
+    if (callback) {
+      const original = callback;
+      callback = (err, result) => {
+        if (err && this.attemptingResume) {
+          // ignore any errors until the change stream has resumed
+          this.onCursorRecreated(() => this.cursor.hasNext(original));
+        } else {
+          original(err, result);
+        }
+      };
+      return this.cursor.hasNext(callback);
```

Comment:
Since we don't need callback again in the code here, could we just open the `.hasNext` callback? I try also to be conscious of [no-param-reassign](https://eslint.org/docs/rules/no-param-reassign).